### DISCUSSION TITLE
Porting SocketIO Server to another language, what resources should I be looking at?

### DISCUSSION THREAD

**Discussion by danielAsaboro** [socketio/socket.io#5041] [Q&A]
Hi,

I'm a junior developer, and we use a Specific Language where I'm interning at.

We are building a platform to teach others about programming in a sandbox environment,

We are focused on using a specific language both on the server and on the client, it works well as a
cross-platform language but it's not been extensively used on the server...so it's not Js

one of the objectives is to use a specific language because that's where we originate from and we want to
show the community that it's possible (my boss is a GDE). 

I love SocketIo and I've been using it since when I picked up Js early last year. 

when we started dealing with real-time events, I convinced my boss to let us 
use socketio since I'm not too fond of the APIs of traditional WebSocket, plus its
batteries included — only to realize no one has written the socket server in the language
 but there are many implementations of the client

This is also a learning experience for me

Beyond just consuming socketio API, how can I learn about how it works so I can reimplement it

I'm looking at something similar to [WebRTC for the Curious](https://webrtcforthecurious.com) for learning the dynamics of webrtc....I'm looking at the docs, but it seems it's like an overview

**Comment by darrachequesne** [comment]
Hi! You can take a look at the details of:

- the Engine.IO protocol: https://socket.io/docs/v4/engine-io-protocol/ (low-level)
- the Socket.IO protocol: https://socket.io/docs/v4/socket-io-protocol/ (high-level, built on top on the Engine.IO protocol)

If something is not clear enough, I'll be glad to answer any question!

**Comment by danielAsaboro** [comment]
Hi, thanks for the help

Before my question, I was lost in the space, and didn't understand how it all fit in.
I believe after the question I was now beginning to get it and how it all fits in.

1. Engine Io
2. Socket IO
3. Engine IO Parser
4. Socket IO Parser
5. SocketIO client
6. Engine IO Client, and
7. Socket IO adapter

I think I now see how they all connect,
and I've made my notes here:

https://github.com/danielAsaboro/socket.io_server/tree/main/note

As my boss has said, I should get the smallest bit working: looks like Engine IO to me

thanks for the resources share, 

if I have any questions I will reach out after I've exhausted all my 'options'

I appreciate the kindness
